### PR TITLE
Kyle feat merchant analytics

### PR DESCRIPTION
**Pull request by Kledin85**
This pull request will have iteration 4 complete still needs some refactoring and check test coverage"

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -194,4 +194,93 @@ def upper_bound(int)
   def top_revenue_earners(merch_num = 20)
     ranked_merchants[0..upper_bound(merch_num)]
   end
+
+  def merchants_with_pending_invoices
+    merchants.all.find_all do |merchant|
+      merchant.invoices.any? do |invoice|
+      !invoice_paid_in_full?(invoice.id) # comeback and refactor
```
Comment (Pocketzs) [inline lib/sales_analyst.rb]: indent line 201 2 spaces after 200

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -194,4 +194,93 @@ def upper_bound(int)
...
+  def max_revenue(merchant_id)
+    item_revenue_hash(merchant_id).max_by do |k, v|
+      v
+    end.last
+  end
+
+  def top_items_by_revenue(merchant_id)
+    item_revenue_hash(merchant_id).find_all do |k, v|
+       v == max_revenue(merchant_id)
+     end
```
Comment (beddings81) [inline lib/sales_analyst.rb]: fix this

File: lib/sales_analyst.rb (hunk 1 above)
Comment (beddings81) [reply lib/sales_analyst.rb]: add indentation on line 201

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -194,4 +194,93 @@ def upper_bound(int)
...
+  def max_quantity(merchant_id)
+    item_quantity_hash(merchant_id).max_by do |k, v|
+      v
+    end.last
+  end
+
+  def top_items_by_quantity(merchant_id)
+    item_quantity_hash(merchant_id).find_all do |k, v|
+       v == max_quantity(merchant_id)
+     end
```
Comment (Pocketzs) [inline lib/sales_analyst.rb]: Fix alignment

File: lib/sales_analyst.rb (hunk 2 above)
Comment (Pocketzs) [reply lib/sales_analyst.rb]: fix alignment

File: spec/sales_analyst_spec.rb [hunk 4]
```diff
@@ -395,4 +395,192 @@
...
+      :items     => './data/test_data/most_sold_item_for_merchant/items.csv',
+      :merchants => './data/test_data/most_sold_item_for_merchant/merchants.csv',
+      :invoices  => './data/test_data/most_sold_item_for_merchant/invoices.csv',
+      :invoice_items => './data/test_data/most_sold_item_for_merchant/invoiceitems.csv',
+      :transactions => './data/test_data/most_sold_item_for_merchant/transactions.csv',
+      :customers => './data/customers.csv'
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.top_items_by_revenue(1).length).to eq (1)
```
Comment (Pocketzs) [inline spec/sales_analyst_spec.rb]: move the (1) next to `eq`

**Review by Pocketzs — CHANGES_REQUESTED** [review]

**Review by KaraJoHo — APPROVED** [review]

**Review by beddings81 — APPROVED** [review]
looks good!

**Review by Pocketzs — APPROVED** [review]

**Pull request merged**